nilearn/decoding/decoder.py

```diff
@@ -445,13 +480,22 @@ def fit(self, X, y, groups=None):
         self.screening_percentile_ = _adjust_screening_percentile(
             self.screening_percentile, self.mask_img_, verbose=self.verbose)
 
+        n_final_features = int(X.shape[1] * self.screening_percentile_
+                               * self.clustering_percentile / 10000)
+        if n_final_features < 50:
+            warnings.warn(
+                "After clustering and screening, the decoding model will "
+                "be trained only on {} features. ".format(n_final_features)
+                + "Consider raising clustering_percentile or "
```

Comment:
detail: you don't need to build several strings here. you can remove the `+` and move the `format` call after the last snippet, ie after `parameters"`
```
assert (
    "abc"
    "def"
    ) == "abcdef"
```